server/users/users.service.ts

```diff
@@ -1,29 +1,70 @@
 import * as bcrypt from 'bcrypt';
-import { CreateUserCommand, UserDto, UpdateUserCommand } from './users.dto';
+import { CreateUserCommand, UpdateSettingCommand, UpdateUserCommand } from './users.dto';
 import HttpException from '../core/exceptions/http.exception';
 import { isEmptyObject } from '../core/util';
 import { Users } from '../../database/models/users';
 import { ensureInputIsClean } from '../core/input-sanitizer';
 import { ensureInputIsEmail, ensureLongerThan } from '../core/validation';
+import { UserFactory } from './users.factory';
+import SportsService from '../sports/sports.service';
+import { UserModel } from './users.model';
+import { UserSettings } from '../../database/models/user_settings';
+import { EventRating } from '../../database/models/event_rating';
+import { SportSubscriptions } from '../../database/models/sport_subscriptions';
+import { recordAnalyticsEvent } from '../core/analytics-event.service';
 
 class UserService {
+	private entitiesToInclude = ["user_setting", "sport_subscriptions"]
+	private sportsService = new SportsService()
 
-	public async getAll(): Promise<UserDto[]> {
-		const users = await Users.findAll();
+	public async getAll(): Promise<UserModel[]> {
+		const users = await Users.findAll({ include: this.entitiesToInclude });
 
-		return users.map(this.mapToDto);
+		return Promise.all(users.map(user => this.FromDatabaseWithSports(user)));
 	}
 
-	public async getById(userId: number): Promise<UserDto> {
-		const user = await Users.findByPk(userId);
+	public async getById(userId: number): Promise<UserModel> {
+		const user = await Users.findByPk(userId, { include: this.entitiesToInclude });
 		if (!user) {
 			throw new HttpException(409, "User not found.");
 		}
 
-		return this.mapToDto(user);
+		const model = await this.FromDatabaseWithSports(user);
+
+		return model;
+	}
+
+	public async getByEmail(email: string): Promise<UserModel> {
+		const user = await Users.findOne({ where: { email: email }, include: this.entitiesToInclude });
+		if (!user) {
+			throw new HttpException(409, `Email ${email} not found.`);
+		}
+
+		const model = UserFactory.FromDatabase(user);
+
+		return model;
+	}
+
+	public async getByPassword(password: string): Promise<UserModel> {
+		const user = await Users.findOne({ where: { password: password }, include: this.entitiesToInclude });
+		if (!user) {
+			throw new HttpException(409, `Password ${password} not found.`);
+		}
+
+		const model = UserFactory.FromDatabase(user);
+
+		return model;
+	}
+
+	private async FromDatabaseWithSports(user: Users) {
+		const sports = await this.sportsService.getAll(UserFactory.FromDatabase(user));
```

Comment:
Services should _return_ domain models, never take them in. 

But since this service call only needs userId, feel free to change it to just need an `userId` instead of full `UserDto`.

Reply:
`sportService.getAll` signature changed in https://github.com/mbrizic/ratethegame/pull/28/commits/b158c0718600a8adeb6bacdcd5d09c7e8ab913e4